tests/mock_crust/utils.rs

```diff
@@ -621,22 +608,134 @@ pub fn add_connected_nodes_until_split(
         prefixes.iter().map(Prefix::bit_count).collect::<Vec<_>>()
     );
 
-    // Clear all event queues and clear the `next_relocation_dst` values.
+    clear_all_event_queues(nodes, |event| match event {
+        Event::NodeAdded(..)
+        | Event::NodeLost(..)
+        | Event::TimerTicked
+        | Event::SectionSplit(..) => (),
+        event => panic!("Got unexpected event: {:?}", event),
+    });
+    clear_relocation_overrides(nodes);
+
+    trace!("Created testnet comprising {:?}", prefixes);
+}
+
+// Add connected nodes to the given prefixes, so adding one extra node in any of the
+// returned sub-prefixes, will trigger a split in the parent prefix.
+pub fn add_connected_nodes_until_one_away_from_split(
+    network: &Network<PublicId>,
+    nodes: &mut Vec<TestNode>,
+    prefixes_to_nearly_split: &[Prefix<XorName>],
+    use_cache: bool,
```

Comment:
maybe don't need this flag?  the function is only be used once, with the flag being false

Reply:
It is consistent with the other similar functions.
It seemed reasonable to keep a consistent interface.